### PR TITLE
Readings table

### PR DESCRIPTION
**Pull request by nbarlowATI**
* Add a new frontend page /sensors/readings that displays results for selected sensor/time-period in a table.
* Restore timeseries-plotting functionality to /sensors/index

### CODE REVIEW

File: dtbase/webapp/app/sensors/templates/readings.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    });
+
+    $('#reportrange').on('apply.daterangepicker', function(ev, picker) {
+        var request_url = "/readings/aranet_trh?range=" + picker.startDate.format('YYYYMMDD') + "-" + picker.endDate.format('YYYYMMDD')
+
+        location.replace(request_url)
+    });
+
+</script>
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/readings.html:145]: I'm wondering whether we should keep js in static/javascript, though this is quite short so probably fine in here.

File: dtbase/webapp/app/sensors/templates/sensors.html
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/sensors.html:1]: The sensor plots look great!

File: dtbase/webapp/app/sensors/templates/readings.html
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/readings.html:1]: Looks really good! Pagination, limiting number of entries on a page and search work well too!

File: dtbase/ingress/ingress_weather.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        "units": "degrees",
+    },
+    "rain": {
+        "name": "rain",
+        "units": "mms",
+    },
+    "icon": {
+        "name": "icon",
+        "units": "",
+    },
```
Comment (mastoffel) [inline dtbase/ingress/ingress_weather.py:55]: This seems to create an extra plot which is empty and has an "icon" y-axis. Can that be deleted or is it a placeholder for something?

File: dtbase/webapp/app/sensors/templates/sensors.html [hunk 3]
```diff
@@ -68,10 +68,54 @@ <h3>Choose sensors and time period</h3>
             </div>
 
           </div>
-          <button style="margin-top: 5px" type="button" onclick="requestTimeSeries(false)">Plot</button>
-          <button style="margin-top: 5px" type="button" onclick="requestTimeSeries(true)">Download</button>
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/sensors.html:72]: The Download doesn't seem to work yet.

File: dtbase/webapp/app/sensors/templates/readings.html [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+		    <th>{{ sd['timestamp'] }}</th>
+		    {% for m in measure_names: %}
+		    <th>{{ '%0.2f' | format(sd[m]|float) }} </th>
+		    {% endfor %}
+		  </tr>
+		  {% endfor %}
+                </tbody>
+              </table>
+	      <form method="post">
+		<input type="submit" value="Download" name="download"/>
```
Comment (mastoffel) [inline dtbase/webapp/app/sensors/templates/readings.html:98]: Also this download doesn't work yet.

**Review by mastoffel — COMMENTED** [review]
Looks great! Happy to merge this, but made a few comments below.

**Comment by mhauru** [comment]
It might take me a few days to find the time to look at this. Is there some particular part on which you'd like my comments?

**Comment by nbarlowATI** [comment]
Ah, sorry @mhauru , it was just force-of-habit that I clicked you!   Don't worry about this - @mastoffel has already had a look (and found my ummm....  "deliberate mistake"  :-P )

**Comment by mastoffel** [comment]
😆 which one was it ?

File: dtbase/ingress/ingress_weather.py (hunk 2 above)
Comment (nbarlowATI) [reply dtbase/ingress/ingress_weather.py:55]: So, "icon" is a string that tells applications what weather icon png to show (we used it in the Unity app for CROP).   I guess chart.js doesn't know how to plot strings.   We could add some logic to exclude string measures from the data that gets passed to the template, but I'm not sure it'd be worth the effort for the moment (though may revisit this...)

File: dtbase/webapp/app/sensors/templates/sensors.html (hunk 3 above)
Comment (nbarlowATI) [reply dtbase/webapp/app/sensors/templates/sensors.html:72]: Good point - I think this is functionality worth having - implemented it now.

File: dtbase/webapp/app/sensors/templates/readings.html (hunk 4 above)
Comment (nbarlowATI) [reply dtbase/webapp/app/sensors/templates/readings.html:96]: And also here!

**Pull request merged**